ReaderApp.js

```diff
@@ -1517,12 +1561,18 @@ class ReaderApp extends React.PureComponent {
       filterIndex = this.state.versionFilterIndex;
     }
     if (!segmentRef) { segmentRef = this.state.segmentRef; }
-    this.state.versionRecentFilters[filterIndex].refList = [segmentRef];
+    
+    const newVersionRecentFilters = [...this.state.versionRecentFilters];
```

Comment:
wrap this logic in setState callback and calculate `newVersionRecentFilters` using prevState